cgrok/grok.go

```diff
@@ -237,24 +237,27 @@ func (match *Match) StartIterator() {
 
 /* Try to advance the iterator. Returns false at the end of the list of groups. */
 func (match *Match) Next() bool {
+	match.sublen = 0
+	match.namelen = 0
+	match.suboffset = 0
 	if C.grok_match_walk_next_offsets(&match.gm, &match.name, &match.namelen, &match.suboffset, &match.sublen) != GROK_OK {
 		return false
 	}
 	match.gname = match.grok.gostringn(match.name, match.namelen)
 	match.gsubstring = ""
 	/* Do some pointer arithmetic to find the index of the match within the string. The match should always
-           be between the beginning of the subject and the end, or an empty string */
+	   be between the beginning of the subject and the end, or an empty string */
 	if int(match.sublen) > 0 {
-		if int(match.suboffset) > -1 && int(match.suboffset + match.sublen) <= len(match.subject) {
-			match.gsubstring = match.subject[int(match.suboffset):int(match.suboffset + match.sublen)]
+		if int(match.suboffset) > -1 && int(match.suboffset)+int(match.sublen) <= len(match.subject) {
+			match.gsubstring = match.subject[int(match.suboffset):int(match.suboffset+match.sublen)]
```

Comment:
Does this line also need individual casts to `int` on the second part of the slice, e.g. `:int(match.suboffset)+int(match.sublen)`